src/pages/cardsPage/CardsPage.jsx

```diff
@@ -0,0 +1,59 @@
+import { React, useState, useEffect } from 'react';
+import { Box, TextField, ThemeProvider } from '@mui/material';
+import { UserCard } from '../../components/UserCard/UserCard';
+import theme from './theme';
+import './CardsPageStyles.css';
+import fetchUserData from '../../api/usersApi';
+import useLocalStorage from '../../customHooks/useLocalStorage';
+
+function CardsPage() {
+
+  const [users, setUsers] = useState([]);
+  const [search, setSearch] = useState('');
+  const [favorites, setFavorites] = useLocalStorage([]);
+
+  useEffect(() => {
+    fetchUserData().then(fetchedData => setUsers(fetchedData));
+  }, []);
+
+  const handleInputChange = (event) => {
+    setSearch(event.target.value);
+  }
+
+  const filterUsers = (users, search) => {
```

Comment:
You don't need this function. Add the logic for the searching in the handleSearch handler.
 Also just manipulate one array of users which are displayed which in your case is the users state. IE you should have setUsers(...something here) in the handler

Reply:
Not sure if its supposed to work the way i did it now, since i needs to refresh the page in order to reset the search, but i found no other way to do it with setUsers(). Would need a few more pointers if that's not right.